src/sentry/event_manager.py

```diff
@@ -142,7 +143,18 @@ def has_pending_commit_resolution(group):
     if recent_group_link is None:
         return False
 
-    return not ReleaseCommit.objects.filter(commit__id=recent_group_link.linked_id).exists()
+    # commit has been released and is not in pending commit state
+    if ReleaseCommit.objects.filter(commit__id=recent_group_link.linked_id).exists():
+        return False
+    else:
+        # check if this commit is a part of a PR
+        if PullRequestCommit.objects.filter(commit__id=recent_group_link.linked_id).exists():
```

Comment:
We should be able to skip this query, since we validate that a `PullRequestCommit` row exists when we join to it below